src/TestFramework/Coverage/XmlReport/FileCodeCoverageProvider.php

```diff
@@ -35,43 +35,16 @@
 
 namespace Infection\TestFramework\Coverage\XmlReport;
 
-use array_key_exists;
-use Infection\TestFramework\Coverage\CoverageFileData;
-use Symfony\Component\Finder\SplFileInfo;
+use Infection\TestFramework\Coverage\CoveredFileData;
 
 /**
  * @internal
  * @final
  */
 class FileCodeCoverageProvider
 {
-    /**
-     * @var array<string, CoverageFileData>|null
-     */
-    private $coverage;
-
-    private $coverageFactory;
-
-    public function __construct(PhpUnitXmlCoverageFactory $coverageFactory)
-    {
-        $this->coverageFactory = $coverageFactory;
-    }
-
-    public function provideFor(SplFileInfo $fileInfo): FileCodeCoverage
+    public function provideFor(CoveredFileData $fileData): FileCodeCoverage
     {
-        if ($this->coverage === null) {
-            $this->coverage = $this->coverageFactory->createCoverage();
-        }
-
-        $filePath = $fileInfo->getRealPath() === false
-            ? $fileInfo->getPathname()
-            : $fileInfo->getRealPath()
-        ;
-
-        if (!array_key_exists($filePath, $this->coverage)) {
-            return new FileCodeCoverage(new CoverageFileData());
-        }
-
-        return new FileCodeCoverage($this->coverage[$filePath]);
+        return new FileCodeCoverage($fileData->retrieveCoverageFileData());
```

Comment:
I'm not quite sure to understand why we need both `FileCodeCoverage` and `CoveredFileData`

Reply:
The reason they are so is because I wanted to keep tests as verbatim as possible. I don't recall other reason so they might be merged later just as well.